source/CIPlanetProgressNode.cpp

```diff
@@ -45,33 +40,25 @@ void PlanetProgressNode::draw(const std::shared_ptr<cugl::SpriteBatch>& batch,
 /**
  * Updates the frame of the planet progress animation
  */
-void PlanetProgressNode::update(float timestep) {
+void PlanetProgressNode::update(float timestep, int lockinLayerSize) {
     if (_planetLayer.isLockedIn) {
         setFrame(CANLOCKIN_END);
         return;
     }
     
     if (_planetLayer.layerSize == 0) {
         _currFrame = LOCKIN_END;
-    } else if (_planetLayer.layerSize == 1) {
-        _currFrame = LOCKIN_ONE;
-    } else if (_planetLayer.layerSize == 2) {
-        _currFrame = LOCKIN_TWO;
-    } else if (_planetLayer.layerSize == 3) {
-        _currFrame = LOCKIN_THREE;
-    } else if (_planetLayer.layerSize == 4) {
-        _currFrame = LOCKIN_FOUR;
     } else {
-        _currFrame = LOCKIN_FIVE;
+        _currFrame = round(((float) _planetLayer.layerSize / (float) lockinLayerSize) * PROGRESS_ARC_END);
     }
     
     _timeElapsed += timestep;
     if (_timeElapsed > SPF) {
         _timeElapsed = 0;
         unsigned int frame = getFrame();
-        if (_planetLayer.layerSize > 4 && _layerNum != 2) {
+        if (_planetLayer.layerSize > (lockinLayerSize - 1) && _layerNum != 2) {
```

Comment:
NIT: you can remove the && _layerNum != 2 part because that was just to prevent to lock in glow from happening on the last layer which previously couldn't be locked in

Reply:
Gotcha. That one's removed 👍